node/src/utils/round_robin.rs

```diff
@@ -143,6 +146,39 @@ where
     }
 }
 
+impl<I, K> WeightedRoundRobin<I, K>
+where
+    I: Debug,
+    K: Copy + Clone + Eq + Hash + IntoEnumIterator + Debug,
+{
+    /// Dump the contents of the queues (`Debug` representation) to a given file.
+    pub async fn debug_dump(&self, file: &mut File) -> Result<(), io::Error> {
+        let mut locks = Vec::new();
+        for kind in K::into_enum_iter() {
+            let queue_guard = self
+                .queues
+                .get(&kind)
+                .expect("missing queue while dumping")
+                .queue
+                .lock()
+                .await;
+
+            locks.push((kind, queue_guard));
+        }
+
+        let mut writer = BufWriter::new(file);
+        for (kind, guard) in locks {
+            let queue = &*guard;
+            writer.write_all(format!("Queue: {:?} ({}) [\n", kind, queue.len()).as_bytes())?;
+            for event in queue.iter() {
+                writer.write_all(format!("\t{:?}\n", event).as_bytes())?;
+            }
+            writer.write_all(b"]\n")?;
+        }
+        Ok(())
```

Comment:
We should have kept the final `flush()` call.

From [the docs](https://doc.rust-lang.org/std/io/struct.BufWriter.html):

> It is critical to call flush before BufWriter<W> is dropped. Though dropping will attempt to flush the contents of the buffer, any errors that happen in the process of dropping will be ignored. Calling flush ensures that the buffer is empty and thus dropping will not even attempt file operations.

Reply:
Done